MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2946,20 +3019,16 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
 #nullable enable
         public static int GetNumberOfThreads(EncodingJobInfo? state, EncodingOptions encodingOptions, string? outputVideoCodec)
         {
-            if (string.Equals(outputVideoCodec, "libvpx", StringComparison.OrdinalIgnoreCase))
-            {
-                // per docs:
-                // -threads    number of threads to use for encoding, can't be 0 [auto] with VP8
-                //             (recommended value : number of real cores - 1)
-                return Math.Max(Environment.ProcessorCount - 1, 1);
-            }
+            // VP8 and VP9 encoders must have their thread counts set.
+            bool mustSetThreadCount = string.Equals(outputVideoCodec, "libvpx", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(outputVideoCodec, "libvpx-vp9", StringComparison.OrdinalIgnoreCase);
 
             var threads = state?.BaseRequest.CpuCoreLimit ?? encodingOptions.EncodingThreadCount;
 
-            // Automatic
             if (threads <= 0)
             {
-                return 0;
+                // Automatically set thread count
+                return mustSetThreadCount ? Math.Max(Environment.ProcessorCount - 1, 1) : 0;
             }
             else if (threads >= Environment.ProcessorCount)
```

Comment:
This is higher than the recommended value for vpx, is it not?

Reply:
This does not change the behavior from before, but no, vpx with `-row-mt` enabled should be able to exploit row-based parallelism during encoding (in addition to any tile-based parallelism).